security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,42 @@
+package crypter
+
+import "errors"
+
+const counterLen = 12
+
+// counter is a 96-bit, little-endian counter.
+type counter struct {
+	value   [counterLen]byte
```

Comment:
Since we're using a 64-bit sequence number, can we use `uint64` instead? This is similar to the C++ version: 

https://github.com/matthewstevenson88/grpc/blob/b4521fa8f145bd319d91d98fcaaed7ad7018d67e/src/core/tsi/s2a/record_protocol/s2a_crypter.cc#L34

Reply:
Done, I initially thought that converting on the fly would be inefficient but I think it is actually more efficient. I've changed it to use uint64.